crates/stress-test/src/main.rs

```diff
@@ -0,0 +1,176 @@
+        parallel,
+        serial_steps,
+        count,
+        timeout,
+    } = Cli::parse();
+    let shim: String = shim.to_string_lossy().into_owned();
+
+    let c8d = mocks::Containerd::new(verbose).await?;
+
+    let shim = c8d.start_shim(shim).await?;
+    let shim = Arc::new(shim);
+
+    let permits = if parallel == 0 {
+        count as _
+    } else {
+        parallel as _
+    };
+    let semaphore = Arc::new(Semaphore::new(permits));
+    let mut tracker: JoinSet<Result<()>> = JoinSet::new();
+
+    let serial_steps = match serial_steps {
+        Step::Create => "c",
+        Step::Start => "cs",
+        Step::Wait => "csw",
+        Step::Delete => "cswd",
+    };
+
+    for _ in 0..count {
+        let shim = shim.clone();
+        let semaphore = semaphore.clone();
+
+        tracker.spawn(async move {
+            let task = shim.task().await?;
+
+            {
+                // take a permit as this saction might have to run serially
+                let _permit = semaphore.acquire().await?;
+                if serial_steps.contains('c') {
+                    task.create(container_output).await?;
+                }
+                if serial_steps.contains('s') {
+                    task.start().await?;
+                }
+                if serial_steps.contains('w') {
+                    task.wait().await?;
+                }
+                if serial_steps.contains('d') {
+                    task.delete().await?;
+                }
+            }
+
+            if !serial_steps.contains('c') {
+                task.create(container_output).await?;
+            }
+            if !serial_steps.contains('s') {
+                task.start().await?;
+            }
+            if !serial_steps.contains('w') {
+                task.wait().await?;
+            }
+            if !serial_steps.contains('d') {
+                task.delete().await?;
+            }
+
+            Ok(())
+        });
+    }
+
+    println!("Waiting for tasks to finish.");
+    println!("Press Ctrl-C to terminate.");
+
+    let mut success = 0;
+    let mut failed = 0;
+    let ping = Arc::new(Notify::new());
+    let _ = async {
+        let ping = ping.clone();
+        let count = count as usize;
+        while let Some(res) = tracker.join_next().await {
+            ping.notify_waiters();
+            match res {
+                Ok(Ok(())) => {
+                    success += 1;
+                    println!(" > {} .. [OK]", count - tracker.len());
+                }
+                Ok(Err(err)) => {
+                    failed += 1;
+                    println!(" > {} .. {err}", count - tracker.len());
+                }
+                Err(err) => {
+                    println!(" > {} .. {err}", count - tracker.len());
+                }
+            }
+        }
+        let _ = shim.shutdown().await;
+        c8d.shutdown().await
+    }
+    .with_watchdog(timeout, ping.clone())
+    .or_ctrl_c()
+    .await;
```

Comment:
```suggestion
    .await?;
```

Reply:
That was intentional, as this would return the error branch if the watchdog it hit, or the user pressed ctrl-c.
Instead of erroring, we track the completed tasks, and report them at the end.